File: src/main/java/com/jcloisterzone/action/BazaarSelectTileAction.java
package com.jcloisterzone.action;

import com.jcloisterzone.game.capability.BazaarItem;
import io.vavr.collection.Set;

public class BazaarSelectTileAction extends AbstractPlayerAction<BazaarItem> {

	boolean noAuction;
	
    public BazaarSelectTileAction(Set<BazaarItem> options, boolean noAuction) {
        super(options);
        this.noAuction = noAuction;
    }
    
    public boolean getNoAuction() {
    	return noAuction;
    }
}

File: src/main/ts/com/jcloisterzone/action/BazaarSelectTileAction.ts
import { AbstractPlayerAction } from "./AbstractPlayerAction.js";

/** Choose which bazaar supply tile to auction (or take, when no-auction). */
export class BazaarSelectTileAction extends AbstractPlayerAction<void> {
  constructor(private readonly noAuction: boolean) {
    super(null);
  }

  getNoAuction(): boolean {
    return this.noAuction;
  }

  override isEmpty(): boolean {
    return false;
  }
}
